# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: layout/base/nsLayoutUtils.h
# Commit: ec44194caefb
# Full Hash: ec44194caefb41931247d18ad886326cf6d89a93
# Author: Boris Chiou <boris.chiou@gmail.com>
# Date: 2024-03-05 09:48:50
# Description:
#   Bug 1870200 - Break the cyclic dependency if any of the desendants uses non-scaling-stroke. r=emilio
#   
#   While we are computing the transform-box:stroke-box, for CSS Transforms or
#   Motion path Transforms, we have to avoid the cyclic dependency not only
#   for the SVG geometry frame itself, but also for the desendants of
# ==============================================================================

diff -r 5c84d062c789 -r ec44194caefb layout/base/nsLayoutUtils.h
--- a/layout/base/nsLayoutUtils.h	Mon Mar 04 20:33:13 2024 +0000
+++ b/layout/base/nsLayoutUtils.h	Mon Mar 04 20:46:05 2024 +0000
@@ -2913,7 +2913,13 @@
 
   // Compute the geometry box for SVG layout. The caller should map the CSS box
   // into the proper SVG box.
-  static nsRect ComputeSVGReferenceRect(nsIFrame*, StyleGeometryBox);
+  // |aMayHaveCyclicDependency| is used for stroke-box to avoid the cyclic
+  // dependency if any of its descendants uses non-scaling-stroke.
+  enum class MayHaveNonScalingStrokeCyclicDependency : bool { No, Yes };
+  static nsRect ComputeSVGReferenceRect(
+      nsIFrame*, StyleGeometryBox,
+      MayHaveNonScalingStrokeCyclicDependency =
+          MayHaveNonScalingStrokeCyclicDependency::No);
 
   // Compute the geometry box for CSS layout. The caller should map the SVG box
   // into the proper CSS box.